org.eclipse.emfcloud.validation/src/main/java/org/eclipse/emfcloud/validation/ValidationFramework.java

```diff
@@ -0,0 +1,204 @@
+		if (validationFilterList.contains(filter)) {
+			validationFilterList.remove(filter);
+		} else {
+			validationFilterList.add(filter);
+		}
+		this.validate();
+	}
+
+	private void readData(String body) throws IOException {
+		ObjectMapper mapper = new ObjectMapper();
+		mapper.registerModule(new ValidationMapperModule());
+		mapper.registerModule(new ValidationResultModule());
+		JsonNode node = mapper.readTree(body);
+		if (node.get("type").asText().equals("validationResult")) {
+			JsonNode responseData = node.get("data");
+			updateRecentValidationResult(jsonToValidationResultList(mapper, responseData));
+		}
+	}
+
+	private void readConstraintList(String body) throws IOException {
+		ObjectMapper mapper = EMFModule.setupDefaultMapper();
+		mapper.registerModule(new ValidationMapperModule());
+		JsonNode node = mapper.readTree(body);
+		if (node.get("type").asText().equals("success")) {
+			JsonNode listOfElements = node.get("data");
+			// Iterate over all Elements
+			Iterator<String> iterElements = listOfElements.fieldNames();
+			while (iterElements.hasNext()) {
+				String elementKey = iterElements.next();
+				HashMap<String, EMFFacetConstraints> featuresMap = new HashMap<>();
+				JsonNode listOfFeatures = listOfElements.get(elementKey);
+				// Iterate over all Features
+				Iterator<String> iterFeature = listOfFeatures.fieldNames();
+				while (iterFeature.hasNext()) {
+					String featureKey = iterFeature.next();
+					JsonNode facets = listOfFeatures.get(featureKey);
+
+					EMFFacetConstraints emfFacetConstraints = new EMFFacetConstraints(
+							mapper.convertValue(facets, Map.class));
+					featuresMap.put(featureKey, emfFacetConstraints);
+				}
+				this.inputValidationMap.put(elementKey, featuresMap);
+			}
+		}
+	}
+
+	public void updateRecentValidationResult(List<ValidationResult> validationResults) {
+		this.recentValidationResult = validationResults;
+		this.changeListener.changed(validationResults);
+	}
+
+	List<ValidationResult> jsonToValidationResultList(ObjectMapper mapper, JsonNode responseData)
+			throws JsonProcessingException {
+		List<ValidationResult> result = new ArrayList<ValidationResult>();
+		if (!isDiagnosisMessage(responseData.get(ValidationMapperModule.CODE).asInt(),
+				responseData.get(ValidationMapperModule.SOURCE).asText())) {
+			if (responseData.get(ValidationMapperModule.SEVERITY).asInt() != 0) {
+				ValidationResult validationResult = mapper.treeToValue(responseData, ValidationResult.class);
+				ValidationFilter filter = new ValidationFilter(validationResult.getDiagnostic().getCode(),
+						validationResult.getDiagnostic().getSource());
+				if (!validationFilterList.contains(filter)) {
+					result.add(validationResult);
+				}
+			}
+		}
+		for (JsonNode diagnosticData : responseData.get("children")) {
+			result.addAll(jsonToValidationResultList(mapper, diagnosticData));
+		}
+		return result;
+	}
+
+	private boolean isDiagnosisMessage(int code, String source) {
```

Comment:
wouldn't it make sense to simply use:
```suggestion
	private boolean isDiagnosisMessage(JsonNode responseData) {
```
?